packages/admin/cms-admin/src/dam/DataGrid/selection/DamMoreActions.tsx

```diff
@@ -26,6 +30,49 @@ export const DamMoreActions = ({ button, transformOrigin, anchorOrigin }: DamMor
         setAnchorEl(null);
     };
 
+    const selectionMapValues = Array.from(damSelectionActionsApi.selectionMap.values());
+    const lengthOfSelectedFiles = selectionMapValues.filter((value) => value === "file").length;
+    const isFolderInSelection = selectionMapValues.some((value) => value === "folder");
+    const onlyFoldersSelected = selectionMapValues.every((value) => value === "folder");
+
+    const snackbarComponent = (
```

Comment:
Please only create this component if it's needed. You can move `snackbarComponent` and `isFolderInSelection` into `onDownloadClick`. This way, the code is nearer to where it's actually used, making it easier to read